push-sdk-swift/AGPushAnalytics.swift

```diff
@@ -0,0 +1,130 @@
+/*
+* JBoss, Home of Professional Open Source.
+* Copyright Red Hat, Inc., and individual contributors
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+import Foundation
+import UIKit
+
+/**
+* Utility class used to send metrics information to the AeroGear UnifiedPush Server when the app is opened due to a Push notification.
+*/
+public class AGPushAnalytics {
+    struct AGPushAnalyticsError {
+        static let AGPushAnalyticsErrorDomain = "AGPushAnalyticsErrorDomain"
+        static let AGNetworkingOperationFailingURLRequestErrorKey = "AGNetworkingOperationFailingURLRequestErrorKey"
+        static let AGNetworkingOperationFailingURLResponseErrorKey = "AGNetworkingOperationFailingURLResponseErrorKey"
+    }
+    /**
+    * Send metrics to the AeroGear Push server when the app is first launched or bring from background to
+    * foreground due to a push notification.
+    *
+    * @param messageId The identifier of this push notification.
+    *
+    * @param completionHandler A block object to be executed when the send metrics operation finishes.
+    * Defaulted to no action.
+    */
+    class public func sendMetricsWhenAppLaunched(launchOptions: [NSObject:AnyObject]?, completionHandler: ((error: NSError? ) -> Void) = {(error: NSError?) in }) {
```

Comment:
I'll rather keep the optional as the appDelegate method. Given that wether it's nil or not is tested within the method. This was part of the discussion we had with @matzew to do all testing with the sdk